modules/apps/object/object-rest-impl/src/main/java/com/liferay/object/rest/internal/vulcan/openapi/contributor/ObjectEntryOpenAPIContributor.java

```diff
@@ -126,28 +133,66 @@ else if (key.contains("objectRelationshipName")) {
 								relatedSchemaName);
 						}
 
-						_addObjectRelationshipPathItem(
-							key, objectRelationship, paths, relatedSchemaName);
+						if (!(Objects.equals(
+								objectRelationship.getType(),
+								ObjectRelationshipConstants.TYPE_ONE_TO_MANY) &&
+							  (objectRelationship.getObjectDefinitionId2() ==
+								  _objectDefinition.getObjectDefinitionId()))) {
+
+							_addObjectRelationshipPathItem(
+								key, objectRelationship, paths,
+								relatedSchemaName);
+						}
 					}
 
-					openAPI.getComponents(
-					).getSchemas(
-					).get(
-						_objectDefinition.getShortName()
-					).getProperties(
-					).put(
-						objectRelationship.getName(),
-						new Schema<Object>() {
-							{
-								setDescription(
-									StringBundler.concat(
-										"Information about the relationship ",
-										objectRelationship.getName(),
-										" can be embedded with ",
-										"\"nestedFields\"."));
-							}
+					Schema<Object> nestedEntitySchema = new Schema<>();
+
+					nestedEntitySchema.setDescription(
+						StringBundler.concat(
+							"Information about the relationship ",
+							objectRelationship.getName(),
+							" can be embedded with 	\"nestedFields\"."));
+
+					if (GetterUtil.getBoolean(
+							PropsUtil.get("feature.flag.LPS-168462")) &&
+						(uriInfo != null)) {
+
+						if (Objects.equals(
```

Comment:
I find this code a bit complex. I think it would be easier to include a new method to get the schema to include for each relationship, something like:

```
if (uriInfo != null) {
	openAPI.getComponents(
	).getSchemas(
	).get(
		_objectDefinition.getShortName()
	).getProperties(
	).put(
		objectRelationship.getName(),
		_getSchema(objectRelationship, relatedSchemaName)
	);
}
```


```
private Schema _getSchema(
	ObjectRelationship objectRelationship, String schemaName) {

	Schema schema = new Schema();

	schema.setDescription(
		StringBundler.concat(
			"Information about the relationship ",
			objectRelationship.getName(), " can be embedded with ",
			"\"nestedFields\"."));
	schema.set$ref(schemaName);

	if (objectRelationship.getObjectDefinitionId1() ==
			_objectDefinition.getObjectDefinitionId()) {

		ArraySchema arraySchema = new ArraySchema();

		arraySchema.setItems(schema);

		return arraySchema;
	}

	return schema;
}
```

Reply:
I miss certain tests for this functionality